### DISCUSSION TITLE
How to choose compiler/VisualStudio git bash should use?

### DISCUSSION THREAD

**Discussion by SpaceCowboy850** [git-for-windows/git#3401] [Q&A]
I have two versions of visual studio installed.  One is the community version and the other is the professional version.  The community version fails at compiling cuda, and I think the professional version will work (due to both being on different versions of visual studio).

However, I cannot figure out how to choose which one git-bash is using, and it is defaulting to the community version.  I've done a bit of googling and can't seem to find how to change this behavior.  Any suggestions?

**Comment by ROGERSM94** [comment]
I'd check your path variable, which is the most likely culprit for these kinds of issues (selecting the wrong version of a program with the same name).

**Reply by SpaceCowboy850** [reply]
When I echo my path at the git-bash prompt, I see nothing about "community" or anything...that was my thought as well prior to posting, but didn't see anything in my environment variables that was a red flag.  I'll look again.

**Comment by mfriedrich74** [comment]
Typically if sln files are used, it's defined by a) what version the
extension is linked to and b) the sln specifies a VS version in it's own
xml (written by the VS edition that was used to save the sln).
Option c) is via environment variables and path, this is what the
vcvars.bat and similar batches from VS use to initialize a specific
compiler environment. You typically do not have this in your global
environment but the batch file is usually called (sourced) in scripts.

If an sln isn't used, e.g. if cmake is used, then cmake rules apply, which
i am net very familiar with.

**Reply by SpaceCowboy850** [reply]
Yes - at the end of the day it got me to thinking this might not be a "git-bash" problem, but a setuptools problem, where setuptools is what is choosing the community edition over the professional edition (just like you indicate that cmake would).  So that is going to be my next avenue of research.  Thanks for the help spitballing this.  If I figure it out I'll post details to follow up.
